xmltojsonlegacy/src/main/java/org/assimbly/xmltojsonlegacy/processor/XmlToJsonProcessor.java

```diff
@@ -0,0 +1,359 @@
+package org.assimbly.xmltojsonlegacy.processor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.log4j.Logger;
+import org.assimbly.xmltojsonlegacy.CustomXmlJsonDataFormat;
+import org.w3c.dom.*;
+
+public class XmlToJsonProcessor {
+
+    static final Logger logger = Logger.getLogger(XmlToJsonProcessor.class);
+
+    private final CustomXmlJsonDataFormat xmlJsonDataFormat;
+
+    private static final String JSON_XML_ATTR_PREFIX = "@";
+    private static final String JSON_XML_TEXT_FIELD = "#text";
+
+    private static final String JSON_XML_ATTR_CLASS = "class";
+    private static final String JSON_XML_ATTR_CLASS_ARRAY = "array";
+    private static final String JSON_XML_ATTR_CLASS_OBJECT = "object";
+
+    private static final String JSON_XML_ATTR_TYPE = "type";
+    private static final String JSON_XML_ATTR_TYPE_NUMBER = "number";
+    private static final String JSON_XML_ATTR_TYPE_BOOLEAN = "boolean";
+
+    public XmlToJsonProcessor(CustomXmlJsonDataFormat xmlJsonDataFormat) {
+        this.xmlJsonDataFormat = xmlJsonDataFormat;
+    }
+
+    public JsonNode convertXmlToJson(
```

Comment:
This method is relative big, maybe put some stuff in a separate method.

Reply:
OK, I already extracted some methods. I think it's better now.